src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -161,11 +160,13 @@ private String storeFile(FileUploadSubmission fileUploadSubmission, MultipartFil
         }
 
         // Note: we can only delete the file, if the file name was changed (i.e. the new file name is different), otherwise this will cause issues
-        if (oldFilePath != null) {
+        Optional<FileUploadSubmission> lastSubmission = participation.findLatestSubmission();
```

Comment:
Why do you need to load this submission now?

Reply:
I don't load it. I just retrieve it from the set.
I need it to delete the file that gets replaced. The oldFilePath variable doesn't hold the value because of the new Upload implementation.